### PR TITLE
remove logic that uses more memory in prepare_attn_masks

### PR DESCRIPTION
**Pull request by libinta**
This has 2 changes
1.  Inside prepare_attn_masks, there is extra code to address upstream gemma3 attn_mask issue(see similar issue reported https://github.com/huggingface/transformers/issues/39389). As this code creates 2 extra tensors, it causes 128k run with OOM. We plan to remove it for now as we don't see any accuracy drop with lmeval test. And plan to fix it either when upstream fixes it or in different PR. Wo this change, the text+image can't support >40k, with this change, can support 108k
3. when warmup multimodal vision bucket, there is no need to execute warmup language model as prompt warmup is separate.

### CODE REVIEW

File: vllm/model_executor/models/gemma3_mm.py [hunk 1]
```diff
@@ -725,29 +725,16 @@ def prepare_attn_masks(
...
-                                               1).unsqueeze(1).unsqueeze(3)
-                ind = torch.arange(seq_len, device=input_ids.device).unsqueeze(
-                    0).unsqueeze(1).unsqueeze(2)
-                img_mask[ind < img_causal] += 1
-                global_attn_mask = torch.where(img_mask == 3, 0,
-                                               global_attn_mask)
+                img_mask1[img_pos1.unsqueeze(1)] += 1
+                img_mask1 = img_mask1.permute(0, 1, 3, 2)
+                img_mask1[img_pos1.unsqueeze(1)] += 1
+                img_mask1 = img_mask1.permute(0, 1, 3, 2)
```
Comment (mswiniarsk) [inline vllm/model_executor/models/gemma3_mm.py]: it seems that img_mask1 is not consumed by any other function, so the code here does not attend to the global_mask at all

File: vllm/model_executor/models/gemma3_mm.py [hunk 2]
```diff
@@ -725,29 +725,16 @@ def prepare_attn_masks(
...
-                img_causal = torch.cat(
-                    (img_causal[:, 0:1] - 1, img_causal[:, :-1]), dim=1)
-                img_causal = img_causal.clamp_(min=0, max=seq_len -
-                                               1).unsqueeze(1).unsqueeze(3)
-                ind = torch.arange(seq_len, device=input_ids.device).unsqueeze(
-                    0).unsqueeze(1).unsqueeze(2)
-                img_mask[ind < img_causal] += 1
-                global_attn_mask = torch.where(img_mask == 3, 0,
-                                               global_attn_mask)
+                img_mask1[img_pos1.unsqueeze(1)] += 1
```
Comment (mswiniarsk) [inline vllm/model_executor/models/gemma3_mm.py]: I see that the difference between non-hpu and hpu is that in hpu we must do: 

```
img_mask1[img_pos1.unsqueeze(1)] += 1
img_mask1 = img_mask1.permute(0, 1, 3, 2)
img_mask1[img_pos1.unsqueeze(1)] += 1
img_mask1 = img_mask1.permute(0, 1, 3, 2)
```

vs non-hpu: 

```
img_mask[:, :, :, img_pos] += 1
img_mask[:, :, img_pos, :] += 1
```

Can you please explain why we can't use the non-hpu for hpu execution as well? Does it cause recompilations or some other performance issue?

**Review by mswiniarsk — APPROVED** [review]
LGTM
